### DISCUSSION TITLE
How can I secure my internet facing trilium instance ?

### DISCUSSION THREAD

**Discussion by hokapefr** [TriliumNext/Trilium#1864] [Q&A]
Hi,

I have installed trilium on Ubuntu 20.04 and bound it on a internet faced subdomain.

Is it possible now to find somewhere in the logs the bad user and password attempts ?

I would like to use something like "fail2ban" to ban the origin IPs.

Regards,
Michel

**Comment by zadam** [comment]
Hi, currently unsuccessful login attempts are not logged. Since it's a good idea, I added logging them.

**Comment by hulmgulm** [comment]
Hi,
I'm running Trilium behind a reverse proxy (nginx). Unfortunately, it logs now the IP of nginx and not the IP of the one who tried to log in. Could you log the IP from X-Forwarded-For or X-Real-IP request header if set?
Thank you,
   Andreas

**Reply by zadam** [reply]
Hi, `X-Forwarded-For` should be respected, if `trustedReverseProxy` in the config is enabled: https://github.com/zadam/trilium/blob/master/config-sample.ini#L29

**Reply by hulmgulm** [reply]
Thank you. That's exactly what I was looking for.

**Comment by Talisker69** [comment]
Hi Alls,
I'm wondering how you can have IPs on the logs ? I also used a nginx reverse proxy and fail2ban so I'm strongly interested.
Thanks in advance.

**Reply by d03j** [reply]
the logs already capture IPs on wrong password events.


```
10:21:26.052 WARNING: Wrong password from [IP redacted], rejecting
```
